### PR TITLE
chore: ⬆️  upgrade clickhouse operator to 0.21.2

### LINKED ISSUE #86: PV reconciliation error causes PV to be outOfSync (requires pruning) on argocd
Hello SigNoZ team.

I am trying to deploy SigNoZ with ArgoCD and I have the following problem:

pv
ArgoCD is trying to delete a PV that they should not manage and is in a constant loop.

crd
CRD immediatly OutOfSync

ArgoCD chart version: 5.5.8 

I found  the following related issues:
https://github.com/Altinity/clickhouse-operator/issues/958
https://github.com/argoproj/argo-cd/issues/3391

### PR DESCRIPTION
**Pull request by prashant-shahi**
Fixes #86, #55, https://github.com/SigNoz/signoz/issues/2630

Also, potential fix for the occasional `clickhouse_operator` invalid creds issue.

_Screenshot:_

Screenshot 2023-06-30 at 12 55 46 AM

Signed-off-by: Prashant Shahi <prashant@signoz.io>

### CODE REVIEW

**Comment by srikanthccv** [comment]
What parts of this are auto-generated changes, and what parts have hand-made changes that need a review?

**Comment by prashant-shahi** [comment]
@srikanthccv most of it is from generated CRD, except the lines using templates like `{{ include "clickhouseOperator.fullname" . }}`

**Comment by prashant-shahi** [comment]
Along with upgrading ClickHouse Operator to 0.21.2, following is done:
- configurations for additional environments for clickhouse operator and metrics server
- configurations for TTL and flush interval for query log, part log and trace log

File: charts/clickhouse/templates/clickhouse-operator/configmaps/etc-files.yaml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+      ################################################
+      ##
+      ## Log parameters
+      ##
+      ################################################
+      logger:
+        logtostderr: "true"
+        alsologtostderr: "false"
+        v: "1"
+        stderrthreshold: ""
```
Comment (srikanthccv) [inline charts/clickhouse/templates/clickhouse-operator/configmaps/etc-files.yaml:286]: I am assuming the single `configmap.yaml` is split into multiple files under `configmaps` without any changes (except ttl templating).

**Review by srikanthccv — APPROVED** [review]

File: charts/clickhouse/templates/clickhouse-operator/configmaps/etc-files.yaml (hunk 1 above)
Comment (prashant-shahi) [reply charts/clickhouse/templates/clickhouse-operator/configmaps/etc-files.yaml:286]: yes, you are right.

**Pull request merged**